spyder/widgets/explorer.py

```diff
@@ -541,11 +545,14 @@ def delete(self, fnames=None):
         multiple = len(fnames) > 1
         yes_to_all = None
         for fname in fnames:
-            yes_to_all = self.delete_file(fname, multiple, yes_to_all)
-            if yes_to_all is not None and not yes_to_all:
-                # Canceled
-                return
-
+            if osp.isdir(fname) and osp.exists(osp.join(fname,'.spyproject')):
+                self.delete_project(fname)
+            else:    
+                yes_to_all = self.delete_file(fname, multiple, yes_to_all)
+                if yes_to_all is not None and not yes_to_all:
+                    # Canceled
+                    break
+                

```

Comment:
These changes must not be here. They need to be placed in `widgets/projects/explorer.py` because this is a functionality that belongs only to projects and not to the file explorer.

In that file you need to define a new `delete` method in `ExplorerTreeWidget`, which will override the `delete` method of `FilteredDirView` (of which `ExplorerTreeWidget` is a subclass of).

Besides, the logic you defined can be improved a bit to make it easier to follow. You could write something like:

```python
if fname == self.root_path:
    self.delete_project(fname)
elif '.spyproject' in fname:
    self.delete_project(fname)
else:
   ...
```

The first condition applies when users try to delete the first entry in the explorer (which corresponds to the project's root directory), and the second one to the `.spyproject` dir (which is only visible on Windows, as we discussed yesterday :-).